pinot-segment-local/src/main/java/org/apache/pinot/segment/local/startree/v2/builder/MultipleTreesBuilder.java

```diff
@@ -104,7 +106,7 @@ public MultipleTreesBuilder(@Nullable List<StarTreeIndexConfig> indexConfigs, bo
     _segmentDirectory = SegmentDirectoryPaths.findSegmentDirectory(indexDir);
     _metadataProperties =
         CommonsConfigurationUtils.fromFile(new File(_segmentDirectory, V1Constants.MetadataKeys.METADATA_FILE_NAME));
-    Preconditions.checkState(!_metadataProperties.containsKey(MetadataKey.STAR_TREE_COUNT), "Star-tree already exists");
+    _existingStarTrees = getExistingStarTrees();
```

Comment:
I see a try catch block on line 109. Will the directory _segment.destroy() handle the cleanup of the new dir we're creating, or should we also be calling _existingStarTrees.cleanupDir() in the catch block?

Reply:
Good catch. Will wrap it in a try catch block.  I think it's good to call _existingStarTrees.cleanOutputDirectory() just to be safe.